include/otf2xx/reference.hpp

```diff
@@ -58,31 +60,28 @@ namespace otf2
  * For each definition should be an own reference type, so the address space is seperated in a
  * typesafe manner.
  *
- * \tparam Type Used to seperate address spaces for different definitions
+ * \tparam Tag Used to seperate address spaces for different definitions
  */
-template <typename Type>
-class reference
+template <typename Tag>
+class reference_impl
 {
 public:
     /**
      * @brief ref_type the underlying type of refernce numbers
      *
      * Mostly uint64_t or uint32_t
      */
-    using ref_type = typename traits::reference_type<Type>::type;
+    using ref_type = typename traits::reference_type<Tag>::type;
 
-    /**
-     * @brief tag_type used to distinguish this id space
-     */
-    using tag_type = Type;
+    using tag_type = Tag;
 
-    reference() = delete;
+    reference_impl() = delete;
```

Comment:
While on it: Remove this and add a default param to `reference_impl(ref_type ref = undefined())` (might need to constexpr the undefined function)

Reply:
No, a undefined reference is something that the user should do explicitly.